web/app/test/ResourceListTest.jsx

```diff
@@ -0,0 +1,87 @@
+import _ from 'lodash';
+import Adapter from 'enzyme-adapter-react-16';
+import CallToAction from '../js/components/CallToAction.jsx';
+import ConduitSpinner from '../js/components/ConduitSpinner.jsx';
+import deployRollup from './fixtures/deployRollup.json';
+import Enzyme from 'enzyme';
+import ErrorBanner from '../js/components/ErrorBanner.jsx';
+import { expect } from 'chai';
+import MetricsTable from '../js/components/MetricsTable.jsx';
+import PageHeader from '../js/components/PageHeader.jsx';
+import { processSingleResourceRollup } from '../js/components/util/MetricUtils.js';
+import React from 'react';
+import { ResourceList } from '../js/components/ResourceList.jsx';
+import { shallow } from 'enzyme';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe('Tests for <ResourceList>', () => {
+  it('displays an error if the api call fails', () => {
+    const msg = 'foobar';
+
+    const component = shallow(
+      <ResourceList
+        data={[]}
+        error={msg}
+        loading={false} />
+    );
+
+    const err = component.find(ErrorBanner);
+    expect(err).to.have.length(1);
+    expect(component.find(PageHeader)).to.have.length(1);
+    expect(component.find(ConduitSpinner)).to.have.length(0);
+    expect(component.find(CallToAction)).to.have.length(1);
+    expect(component.find(MetricsTable)).to.have.length(0);
+
+    expect(err.props().message).to.equal(msg);
+  });
+
+  it('shows a loading spinner', () => {
+    const component = shallow(
+      <ResourceList
+        data={[]}
+        loading={true} />
+    );
+
+    expect(component.find(ErrorBanner)).to.have.length(0);
+    expect(component.find(PageHeader)).to.have.length(1);
+    expect(component.find(ConduitSpinner)).to.have.length(1);
```

Comment:
This is minor, but I don't think this test would pass against the current conduit master. Specifically, in the existing implementation, if the `ConduitSpinner` component is present, the `PageHeader` component is not. So to test the existing behavior, these expectations should be:

```
    expect(component.find(ErrorBanner)).to.have.length(0);
    expect(component.find(PageHeader)).to.have.length(0);
    expect(component.find(ConduitSpinner)).to.have.length(1);
    expect(component.find(CallToAction)).to.have.length(0);
    expect(component.find(MetricsTable)).to.have.length(0);
```

I'm ok with changing the behavior to display both the page header and the spinner simultaneously, but I want to make sure we're intentional about it. For instance, if we do go this route, then we should update the `Namespace` and `ServiceMesh` components as well, to make sure we're consistently displaying spinner + page header throughout.

Reply:
Nope, you're totally right. This is my bad on reading the code again. Once the jest stuff lands, I want to spend some time getting the tests a little bit better. I've realized I am relying on them to yell at me a little too much given their current state.